tests/beta/test_default_tensor_tf.py

```diff
@@ -637,7 +637,7 @@ class TestHybridInterfaceIntegration:
     )
 
     @pytest.fixture
-    def cost(self, interface, torch_support, rep):
+    def cost(self, diff_method, interface, torch_support, rep):
```

Comment:
It took me a while to understand this change, because github wasn't showing the relevant place where this change was needed. Where was it getting `diff_method` from before?

Reply:
Also, there are a number of tests of the form 
```
@pytest.mark.parametrize("interface", "xyz")
test_xyz_interface(..., interface, ...):
```
@josh146 explained to me that this was necessary due to a short-lived bug in `pytest`, which seems now to have been fixed. @antalszava could you update these tests to just be of the following form?:
```
def test_xyz_interface(...):
   interface = "xyz"
```

Reply:
@co9olguy, the suggested change won't work, since we are using indirect fixture injection here. We can remove `interface` from the signature, but the parametrize decorator must remain.

E.g.,

```python
@pytest.mark.parametrize("interface", ["xyz"])
test_xyz_interface(circuit, tol): # no interface in this signature
```

This is because the `circuit` fixture is parametrized by `interface`.

Reply:
The `"diff_method"` fixture was added because the following is envisioned:
```python
    @pytest.mark.parametrize("interface", ["tf"])
    @pytest.mark.parametrize("diff_method", ["device", "classical"])
    def test_tf_interface(self, cost, interface, diff_method, tol):
```
For now we have to split this case up into a `test_tf_interface_device_diff` and `test_tf_interface_classical_diff` case due to #620.

One takes `    @pytest.mark.parametrize("diff_method", ["device"])
` the other `    @pytest.mark.parametrize("diff_method", ["classical"])
`.

Reply:
Once https://github.com/XanaduAI/pennylane/pull/631 is effective which attempts to solve #620, I'd be happy to change these tests so that there is a single test case for `"device"` and  `"classical"` or otherwise include refactors, let me know @co9olguy

Reply:
@antalszava this feature is slotted for the next release. Now that #631 is merged, go ahead and clean up the tests as needed

Reply:
Thanks @co9olguy! :slightly_smiling_face: I've pushed a commit, let me know if you'd rather I changed something further